laws/applicative.js

```diff
@@ -1,36 +1,26 @@
 'use strict';
 
-const {identity, thrush} = require('fantasy-combinators');
 const {of, ap} = require('..');
 
-/**
-
-### Applicative
-
-1. `v.ap(A.of(x => x))` is equivalent to `v` (identity)
-2. `A.of(x).ap(A.of(f))` is equivalent to `A.of(f(x))` (homomorphism)
-3. `A.of(y).ap(u)` is equivalent to `u.ap(A.of(f => f(y)))` (interchange)
-
-**/
-
-const identityʹ = T => eq => x => {
-  const a = T[of](x)[ap](T[of](identity));
-  const b = T[of](x);
-  return eq(a, b);
-};
-
-const homomorphism = T => eq => x => {
-  const a = T[of](x)[ap](T[of](identity));
-  const b = T[of](identity(x));
-  return eq(a, b);
-};
-
-const interchange = T => eq => x => {
-  const u = T[of](identity);
-
-  const a = T[of](x)[ap](u);
-  const b = u[ap](T[of](thrush(x)));
-  return eq(a, b);
-};
-
-module.exports = {identity: identityʹ, homomorphism, interchange};
+const identity = eq => (A, v) => {
+  return eq(
+    v[ap](A[of](x => x)),
+    v
+  );
+}
+
+const homomorphism = eq => (A, x, f) => {
```

Comment:
Thinking a bit more `(A, eq) => (x, f) =>` makes sense. We can make it that all functions take type representative and `eq` function for our type, and return a function that takes rest of dependencies for law equations.

Some examples:

```js
// passing M for unification even though we don't need it for law equation
Monoid.rightIdentity = (M, eq) => (m) => ...

// passing T as usual even though it's not needed here, 
// other type representative are passed 
// in the second chunk of arguments as other dependencies usually passed 
Traversable.composition = (T, eq) => (F, G, u) => ...
```

This way users may automatically partially apply all functions to specialize them for the type they want to test.